emmet-core/emmet/core/openmm/tasks.py

```diff
@@ -0,0 +1,217 @@
+        None, description="List of potential energies"
+    )
+
+    kinetic_energy: Optional[list[float]] = Field(
+        None, description="List of kinetic energies"
+    )
+
+    total_energy: Optional[list[float]] = Field(
+        None, description="List of total energies"
+    )
+
+    temperature: Optional[list[float]] = Field(None, description="List of temperatures")
+
+    volume: Optional[list[float]] = Field(None, description="List of volumes")
+
+    density: Optional[list[float]] = Field(None, description="List of densities")
+
+    elapsed_time: Optional[float] = Field(
+        None, description="Elapsed time for the calculation"
+    )
+
+    @classmethod
+    def from_directory(
+        cls,
+        dir_name: Path | str,
+        state_file_name: str,
+        dcd_file_name: str,
+        elapsed_time: Optional[float] = None,
+        steps: Optional[int] = None,
+        state_interval: Optional[int] = None,
+    ) -> CalculationOutput:
+        """Extract data from the output files in the directory."""
+        state_file = Path(dir_name) / state_file_name
+        column_name_map = {
+            '#"Step"': "steps",
+            "Potential Energy (kJ/mole)": "potential_energy",
+            "Kinetic Energy (kJ/mole)": "kinetic_energy",
+            "Total Energy (kJ/mole)": "total_energy",
+            "Temperature (K)": "temperature",
+            "Box Volume (nm^3)": "volume",
+            "Density (g/mL)": "density",
+        }
+        state_is_not_empty = state_file.exists() and state_file.stat().st_size > 0
+        state_steps = state_interval and steps and steps // state_interval or 0
+        if state_is_not_empty and (state_steps > 0):
+            data = pd.read_csv(state_file, header=0)
+            data = data.rename(columns=column_name_map)
+            data = data.filter(items=column_name_map.values())
+            data = data.iloc[-state_steps:]
+            attributes = data.to_dict(orient="list")
+        else:
+            attributes = {name: None for name in column_name_map.values()}
+            state_file_name = None  # type: ignore[assignment]
+
+        dcd_file = Path(dir_name) / dcd_file_name
+        dcd_is_not_empty = dcd_file.exists() and dcd_file.stat().st_size > 0
+        dcd_file_name = dcd_file_name if dcd_is_not_empty else None  # type: ignore
+
+        return CalculationOutput(
+            dir_name=str(dir_name),
+            elapsed_time=elapsed_time,
+            dcd_file=dcd_file_name,
+            state_file=state_file_name,
+            **attributes,
+        )
+
+
+class Calculation(BaseModel):
+    """All input and output data for an OpenMM calculation."""
+
+    dir_name: Optional[str] = Field(
+        None, description="The directory for this OpenMM calculation"
+    )
+
+    has_openmm_completed: Optional[Union[TaskState, bool]] = Field(
+        None, description="Whether OpenMM completed the calculation successfully"
+    )
```

Comment:
This field is great to have. I'm not (yet) familiar enough with atomate2 to know if there's a more standard way to report whether the executable has finished. For example you have `state` in `ClassicalMDTaskDocument`

If there is not a common standard emerging yet, I'd say it's fine as is.